src/hotspot/share/c1/c1_Compilation.cpp

```diff
@@ -646,6 +647,15 @@ void Compilation::notice_inlined_method(ciMethod* method) {
 
 void Compilation::bailout(const char* msg) {
   assert(msg != nullptr, "bailout message must exist");
+  // record the bailout for hserr envlog
+  if (msg != nullptr) {
```

Comment:
How can it be nullptr? All callers pass some message.

Reply:
What if I create a new call like 
```
char* myMsg = nullptr;
 . . .
bailout(myMsg);
```
and, due to complicated logic, `myMsg` is not assigned a value in all cases? General topic: future-proof code.

Reply:
OK, with that knowledge and without the check, we crash when we try to log the null message. Otherwise, we crash or fail sometime later - or some strange things happen.

If we follow the "fail early" principle, the additional check should not be there. We could as well convert the assert into a guarantee to enforce a "planned" crash.